lib/active_record/with_eager_group.rb

```diff
@@ -9,7 +9,9 @@ def exec_queries
     end
 
     def eager_group(*args)
-      check_if_method_has_arguments!('eager_group', args)
```

Comment:
This method would flatten! args, and cause test case fail.
`posts = Post.eager_group([:comments_average_rating_by_author, students[0], true])`

Reply:
I think it is not necessary to flatten args, just need to focus on raise exception if args is empty.

Reply:
@yakjuly I'd like to keep `check_if_method_has_arguments!`, which is the behavior what active_record used in `includes`, `eager_loads`, etc.

Reply:
@flyerhzm eager_group usage is different than `includes` and `eager_loads`.
eager_group allows scope with arguments, `includes` and `eager_loads` does not.
In project, engineer often have a requirement that return the counts for current user or current context accessible resources. In this case, we do need to pass context or user information to the scope method.

```
posts_1 = Post.eager_group([:comments_average_rating_by_author, students[0], true])
posts_2 = Post.eager_group(:comments_average_rating, :approved_comments_count)
```

In above examples `posts_1`, after arguments be flattened, the 2nd argument `students[0]` will be considered as an eager_group definition and cause exception.
To support eager_group with scope arguments, I think we need to make a trade off not to support below pattern.
```
posts_2 = Post.eager_group([:comments_average_rating, :approved_comments_count])
```

If developers need to pass an array, there is a walk around, developers still able to use `Post.eager_group(*[:comments_average_rating, :approved_comments_count])` to get same thing.

If you still want to flatten arguments, I would suggest to wrap `[:comments_average_rating_by_author, students[0], true]` as an object something like `eager_group(EagerGroupWithScope.new(:comments_average_rating_by_author, students[0], true), ...)`

Reply:
After using `check_if_method_has_arguments!`, eager_group gem can not load different type eager group definitions at same time.

e.g.
```
# ArgumentError:
#       wrong number of arguments (given 3, expected 0)

it 'eager_group multiple aggregate definitions' do
  students = Student.all
  posts = Post.eager_group(:approved_comments_count,[:comments_average_rating_by_author, students[0], true])
  #comments_average_rating_by_author
  post = posts[0]
  object_id1 = post.instance_variable_get('@comments_average_rating_by_author').object_id
  object_id2 = post.comments_average_rating_by_author.object_id
  object_id3 = post.comments_average_rating_by_author.object_id
  expect(object_id1).to eq object_id2
  expect(object_id1).to eq object_id3
  # approved_comments_count
  object_id1 = post.instance_variable_get('@approved_comments_count').object_id
  object_id2 = post.approved_comments_count.object_id
  object_id3 = post.approved_comments_count.object_id
  expect(object_id1).to eq object_id2
  expect(object_id1).to eq object_id3
end
```